src/www/ui/api/Controllers/UploadController.php

```diff
@@ -650,4 +651,73 @@ private function areAgentsScheduled($uploadId, $agents, $response)
     }
     return true;
   }
+
+  /**
+   * Set permissions for a upload in a folder for different groups
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function setUploadPermissions($request, $response, $args)
+  {
+    $returnVal = null;
+    // checking if the scheduler is running or not
+    $commu_status = fo_communicate_with_scheduler('status', $response_from_scheduler, $error_info);
+    if ($commu_status) {
+      // Initialising upload-permissions plugin
+      global $container;
+      $restHelper = $container->get('helper.restHelper');
+      $uploadPermissionObj = $restHelper->getPlugin('upload_permissions');
+
+      $dbManager = $this->dbHelper->getDbManager();
+      // parsing the request body
+      $reqBody = $this->getParsedBody($request);
+
+      $folder_pk = intval($reqBody['folderId']);
+      $upload_pk = intval($args['id']);
+      $upload = $this->uploadAccessible($this->restHelper->getGroupId(), $upload_pk);
+      if ($upload !== true) {
+        return $response->withJson($upload->getArray(), $upload->getCode());
+      }
+      $allUploadsPerm = $reqBody['allUploadsPermission'] ? 1 : 0;
+      $newgroup = intval($reqBody['newGroupId']);
+      $newperm = intval($this->getEquivalentValueForPermission($reqBody['newPermission']));
+      $public_perm = isset($reqBody['publicPermission']) ? $this->getEquivalentValueForPermission($reqBody['publicPermission']) : -1;
+
+      $query = "SELECT perm, perm_upload_pk FROM perm_upload WHERE upload_fk=$upload_pk and group_fk=$newgroup;";
+      $result = $dbManager->getSingleRow($query, [], "");
```

Comment:
Query is prone to SQL injection. Use safer methods.
```suggestion
      $query = "SELECT perm, perm_upload_pk FROM perm_upload WHERE upload_fk=$1 and group_fk=$2;";
      $result = $dbManager->getSingleRow($query, [$upload_pk, $newgroup], __METHOD__.".getOldPerm");
```

Reply:
Sure!